### PR TITLE
Home and user settings pages

### LINKED ISSUE #42: Create user profile page
User's profile page should display the following things
 - [ ] Active games the user is participating in
 - [ ] Orgs the user is part of
 - [ ] settings
   - Change name
   - Change profile picture
   - Change password
   - Log out

### PR DESCRIPTION
**Pull request by hrice9**
fixes #42

### CODE REVIEW

File: HVZ.Web/Shared/User Settings/PasswordChange.razor [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        var result = await UserManager.ChangePasswordAsync(appUser, CurrentPassword!, NewPassword!);
+
+        if (result.Succeeded)
+        {
+            // Display success notification
+            // Email the user
+            // Close the modal
+            changeSuccess = true;
+            return;
+        }
```
Comment (Mogiiii) [inline HVZ.Web/Shared/User Settings/PasswordChange.razor:145]: are these comments TODO?

**Review by Mogiiii — CHANGES_REQUESTED** [review]

File: HVZ.Web/Shared/User Settings/PasswordChange.razor (hunk 1 above)
Comment (hrice9) [reply HVZ.Web/Shared/User Settings/PasswordChange.razor:145]: they're eventual TODOs - I don't have the time to write the email templates and the JS

Comment (Mogiiii) [reply HVZ.Web/Shared/User Settings/PasswordChange.razor:145]: create an issue for tracking

File: HVZ.Web/Shared/Ui Helpers/ActiveGameLink.razor [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    string GetStatusDecorator()
+    {
+        if (ActiveGame is null)
+        {
+            // Log Error
+            return string.Empty;
+        }
+        // TODO: Update with game paused/not started
```
[the remark this answers is not in the document]
Comment (hrice9) [reply HVZ.Web/Shared/Ui Helpers/ActiveGameLink.razor:69]: yes

File: HVZ.Web/Pages/UserSettings.razor [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+@inject IJSRuntime JS
+@using HVZ.Web.Identity.Models
+@using HVZ.Web.Shared.Ui_Helpers
+@using Microsoft.AspNetCore.Identity
+@using System.ComponentModel.DataAnnotations
+@using HVZ.Web.Shared.User_Settings
+
+@implements IDisposable
+
+@page "/settings"
```
[the remark this answers is not in the document]
Comment (hrice9) [reply HVZ.Web/Pages/UserSettings.razor:15]: the nav bar is currently set up to only look at the first segment of the route when determining whether to highlight an item on the nav bar

File: HVZ.Web/Shared/Ui Helpers/PlayerGameRole.razor [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        {
+            playerGameRole = eventArgs.Role.ToString();
+            InvokeAsync(StateHasChanged);
+        }
+    }
+
+    void PlayerTagged(object? sender, TagEventArgs eventArgs)
+    {
+        
+    }
```
[the remark this answers is not in the document]
Comment (hrice9) [reply HVZ.Web/Shared/Ui Helpers/PlayerGameRole.razor]: this is unused

File: HVZ.Web/Shared/Ui Helpers/PlayerGameRole.razor [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        {
+            // Log Error
+            return;
+        }
+
+        var player = await GameRepo.FindPlayerByGameId(GameId, PlayerId);
+
+        if (player is null)
+        {
+            // Log Error
```
[the remark this answers is not in the document]
Comment (hrice9) [reply HVZ.Web/Shared/Ui Helpers/PlayerGameRole.razor]: this is unused

File: HVZ.Web/Shared/Ui Helpers/ActiveGameLink.razor (hunk 2 above)
[the remark this answers is not in the document]
Comment (Mogiiii) [reply HVZ.Web/Shared/Ui Helpers/ActiveGameLink.razor:69]: lets wait on game changes to be ready so we can complete this if possible.

File: HVZ.Web/Pages/UserSettings.razor (hunk 3 above)
[the remark this answers is not in the document]
Comment (Mogiiii) [reply HVZ.Web/Pages/UserSettings.razor:15]: I'd like to improve that soon, but for now we can keep this as-is.

File: HVZ.Web/Shared/Ui Helpers/ActiveGameLink.razor (hunk 2 above)
[the remark this answers is not in the document]
Comment (hrice9) [reply HVZ.Web/Shared/Ui Helpers/ActiveGameLink.razor:69]: #130

**Review by Mogiiii — APPROVED** [review]

**Pull request merged**